tests/integration/test_dynamodb.py

```diff
@@ -433,66 +439,70 @@ def test_more_than_20_global_secondary_indexes(
+        )
         snapshot.match("PutFirstItem", response)
 
         # now the same data is present so when we pass return values as 'ALL_OLD'
         # it should give us attributes
-        response = table.put_item(Item=item1, ReturnValues="ALL_OLD")
+        response = aws_client.dynamodb.put_item(
+            TableName=TEST_DDB_TABLE_NAME, Item=item1, ReturnValues="ALL_OLD"
+        )
         snapshot.match("PutFirstItemOLD", response)
 
         # now a previous version of data is present, so when we pass return
         # values as 'ALL_OLD' it should give us the old attributes
-        response = table.put_item(Item=item1b, ReturnValues="ALL_OLD")
+        response = aws_client.dynamodb.put_item(
+            TableName=TEST_DDB_TABLE_NAME, Item=item1b, ReturnValues="ALL_OLD"
+        )
         snapshot.match("PutFirstItemB", response)
 
         # we do not have any same item as item2 already so when we add this by default
         # return values is set to None so no Attribute values should be returned
-        response = table.put_item(Item=item2)
+        response = aws_client.dynamodb.put_item(TableName=TEST_DDB_TABLE_NAME, Item=item2)
         snapshot.match("PutSecondItem", response)
 
         # in this case we already have item2 in the table so on this request
         # it should not return any data as return values is set to None so no
         # Attribute values should be returned
-        response = table.put_item(Item=item2)
+        response = aws_client.dynamodb.put_item(TableName=TEST_DDB_TABLE_NAME, Item=item2)
         snapshot.match("PutSecondItemReturnNone", response)
 
         # cleanup
-        table.delete()
+        aws_client.dynamodb.delete_table(TableName=TEST_DDB_TABLE_NAME)
 
     @pytest.mark.aws_validated
-    def test_empty_and_binary_values(self, dynamodb, snapshot, aws_client):
+    def test_empty_and_binary_values(self, snapshot, aws_client):
         table_name = f"table-{short_uid()}"
         resources.create_dynamodb_table(
             table_name=table_name, partition_key=PARTITION_KEY, client=aws_client.dynamodb
         )
-        table = dynamodb.Table(table_name)
 
         # items which are being used to put in the table
-        item1 = {PARTITION_KEY: "id1", "data": ""}
-        item2 = {PARTITION_KEY: "id2", "data": b"\x90"}
+        item1 = {PARTITION_KEY: {"S": "id1"}, "data": {"S": ""}}
+        item2 = {PARTITION_KEY: {"S": "id2"}, "data": {"B": b"\x90"}}
 
-        response = table.put_item(Item=item1)
+        response = aws_client.dynamodb.put_item(TableName=table_name, Item=item1)
         snapshot.match("PutFirstItem", response)
 
-        response = table.put_item(Item=item2)
+        response = aws_client.dynamodb.put_item(TableName=table_name, Item=item2)
         snapshot.match("PutSecondItem", response)
 
         # clean up
-        table.delete()
+        aws_client.dynamodb.delete_table(TableName=table_name)
```

Comment:
nit: might be good while going through here anyway to do this in a safe manner (e.g. using the cleanups fixture instead)

Reply:
I have a TODO note to convert the constants on top of this file into fixture which will take care of this :) I'll address these in follow up PRs

https://github.com/localstack/localstack/pull/8674/files/5701d33c5344d96114f3bdc6372c299ba8a7d1cd#diff-674a25994f44ac7d1dd2bdaff60e30cb9eb9c4b895c36d56b5c33b0a9bcfe7e7R27

Reply:
See https://github.com/localstack/localstack/pull/8747/